### PR TITLE
Feature -> configurable dashboard path

### PR DESCRIPTION
**Pull request by MR-MKZ**
> [!Note] 
> currently, It's just available for dev mode :/

- [x] #71 

image

### CODE REVIEW

**Comment by erfjab** [comment]
Hi, good pr my friend. please fix your git.

**Comment by erfjab** [comment]
> currently, It's just available for dev mode :/


What do you mean?

**Comment by MR-MKZ** [comment]
> What do you mean?

I mean when user use docker for panel it doesn't work because it need to build panel frontend with new path and docker container doesn't have npm to build.
I tried to install node js on container but I couldn't even build the old image and container package manager returned ign when tried to install packages :/

File: .env.example [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
+# SUDO_PASSWORD="admin"
 
 # UVICORN_UDS: "/run/marzban.socket"
-# UVICORN_SSL_CERTFILE = "/var/lib/marzban/certs/example.com/fullchain.pem"
-# UVICORN_SSL_KEYFILE = "/var/lib/marzban/certs/example.com/key.pem"
+# UVICORN_SSL_CERTFILE="/var/lib/marzban/certs/example.com/fullchain.pem"
+# UVICORN_SSL_KEYFILE="/var/lib/marzban/certs/example.com/key.pem"
 
+# To change default dashboard path. Don't forget / at the start and the end of path.
+DASHBOARD_PATH=dashboard
```
Comment (M03ED) [inline .env.example:15]: This should be a comment in example file, also you forgot to use / in start and end and " for string

File: build_dashboard.sh [hunk 2]
```diff
@@ -1,3 +1,14 @@
-cd `dirname $0`/app/dashboard
-VITE_BASE_API=/api/ npm run build --if-present -- --outDir build --base '/dashboard/'
-cp ./build/index.html ./build/404.html
\ No newline at end of file
+#!/usr/bin/env bash
```
Comment (M03ED) [inline build_dashboard.sh:1]: Is it better to get path as input instead of reading env file again.

**Review by M03ED — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (MR-MKZ) [reply .env.example:15]: / at start and end will set automatically and ok I add " for string and make this env as a comment

File: build_dashboard.sh (hunk 2 above)
Comment (MR-MKZ) [reply build_dashboard.sh:1]: Hmm, that's a good idea💡

**Comment by erfjab** [comment]
You forget / 😅

**Comment by MR-MKZ** [comment]
> You forget / 😅
Where?

**Comment by erfjab** [comment]
> Where?

No, I read it again, it's OK.

**Comment by MR-MKZ** [comment]
I found some bugs and I need to fix them before merge so I make this pr draft for a short time to fix bugs.

**Comment by MR-MKZ** [comment]
Structure of this feature has some problems and I have no idea to how fix it

**Pull request closed without merging**